src/Build.UnitTests/Construction/ElementLocation_Tests.cs

```diff
@@ -363,6 +363,7 @@ public void SaveReadOnly1()
             {
                 var doc = new XmlDocumentWithLocation(loadAsReadOnly: true);
                 doc.Load(_pathToCommonTargets);
+                Assert.True(doc.IsReadOnly);
```

Comment:
This is testing that we get an InvalidOperationException on save, you should add a new unit test that just tests the IsReadOnly property and not do these checks in the SaveReadOnly tests.

Reply:
This PR adds dedicated tests for the IsReadOnly property. The additional asserts I added hare are asserting that these test cases are really operating on a read-only object.